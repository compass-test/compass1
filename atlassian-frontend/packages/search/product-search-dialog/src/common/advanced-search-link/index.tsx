export { AdvancedSearchLink } from './advanced-search-link';
export {
  AdvancedSearchContent,
  CenteredAdvancedSearchGroup,
  StyledAdvancedSearch,
} from './advanced-search-link.styled';

export { default as MultiSiteAdvancedSearchLink } from './multi-site-advanced-search-link';
