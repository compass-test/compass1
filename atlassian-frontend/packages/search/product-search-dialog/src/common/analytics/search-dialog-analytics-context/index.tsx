export { SearchDialogAnalyticsContext } from './search-dialog-analytics-context';
