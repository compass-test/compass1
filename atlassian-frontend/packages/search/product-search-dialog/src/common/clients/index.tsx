import * as CommonTypes from './common-types';

export { AggregatorClient } from './base-search-client';
export { CollaborationGraphClient } from './collaboration-graph-client';
export { responseErrorToError } from './error-handler';
export { extractSiteUrl } from './multi-site-utils';
export { timed } from './timing';

export type ModelParam = CommonTypes.ModelParam;
export type SearchContext = CommonTypes.SearchContext;
export type AggregatorConfig = import('./base-search-client').AggregatorConfig;
export type ProductsPermissionsResponse<
  T
> = import('./base-search-client').ProductsPermissionsResponse<T>;
export type CollaborationGraphConfig = import('./collaboration-graph-client').CollaborationGraphConfig;
export type ABTest = CommonTypes.ABTest;
export type ContainerType = CommonTypes.ContainerType;
export type ScopedAggregatorResponse<T> = CommonTypes.ScopedAggregatorResponse<
  T
>;
export type CollaborationGraphResponse<
  T
> = CommonTypes.CollaborationGraphResponse<T>;
export type CollaborationGraphContainer = CommonTypes.CollaborationGraphContainer;
export type SharedClient = CommonTypes.SharedClient;

// Multi site
export type Site = CommonTypes.Site;
