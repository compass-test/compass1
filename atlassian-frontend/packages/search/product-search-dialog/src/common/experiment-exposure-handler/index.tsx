export { default as ExperimentExposureHandler } from './experiment-exposure-handler';
