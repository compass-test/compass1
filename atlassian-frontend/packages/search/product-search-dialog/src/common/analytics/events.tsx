/**
 * Note all events fired include the searchSessionId, experiment-id, queryHash, queryVersion, wordCount, resultCount where applicable
 */

import {
  GasPayload,
  GasScreenEventPayload,
} from '@atlaskit/analytics-gas-types';
import { ABTest } from '../clients';
import { ProductFilterType } from '../../common/clients/common-types';
import { FilterOptionSource } from '../../common/filters/types';
import { Products } from '../../common/product-context';
import { OptionData, Option } from '@atlaskit/user-picker/types';

export const DEFAULT_GAS_CHANNEL = 'fabric-elements';

export enum Trigger {
  SHORTCUT = 'shortcut',
  RETURN = 'return',
  CLICK = 'click',
  TEXT_ENTERED = 'textEntered',
}

export enum AdvancedSearchLinkSubjectId {
  CONFLUENCE = 'confluenceAdvancedSearchLink',
  CONFLUENCE_PEOPLE = 'confluencePeopleSearchLink',
  JIRA_ISSUES = 'jiraIssuesSearchLink',
  JIRA_BOARDS = 'jiraBoardsSearchLink',
  JIRA_FILTERS = 'jiraFiltersSearchLink',
  JIRA_PROJECTS = 'jiraProjectsSearchLink',
  JIRA_PLANS = 'jiraPlansSearchLink',
  JIRA_PEOPLE = 'jiraPeopleSearchLink',
  NO_RESULTS = 'noResultsAdvancedSearchLink',
}

export interface NonPrivacySafeContext {
  query: string;
}

export interface FiltersInfo {
  applied: {
    id: string;
    index: number;
    source: FilterOptionSource;
  }[];
  recommendedIds: {
    id: string;
    source: FilterOptionSource;
  }[];
}

export interface FiltersAnalyticsContext {
  container?: FiltersInfo;
  contributor?: FiltersInfo;
  projects?: FiltersInfo;
  assignees?: FiltersInfo;
  binaryStatusCategories?: FiltersInfo;
}

export interface AdvancedSearchAnalyticsProps {
  actionSubjectId: string;
  isLoading: boolean;
  newTab: boolean;
  trigger: Trigger;
}

export interface MultiSiteAdvancedSearchAnalyticsProps {
  trigger: Trigger;
  actionSubjectId: string;
  isLoading: boolean;
  newTab: boolean;
  destinationId: string;
  index: number;
}

export interface MoreFiltersAnalyticsProps {
  trigger: Trigger;
  selectedFiltersCount: number;
}

export interface TabSelectedProps {
  tabName: Products;
}

const BASE_ANALYTICS_SOURCE = 'searchDialog';

export type ContextPayload = Partial<{
  abTest: ABTest;
  searchSessionId: string;
  queryHash: string;
  queryLength: number;
  queryVersion: number;
  wordCount: number;
  frontend: string;
  collabGraphSessionId?: string;
}>;

export type GasPayloadWithContext = GasPayload;

/**
 * Custom GasPayload type with restrictions so we don't include common attributes that will be added as part of AnalyticContexts
 */
export type LimitedGasPayload = GasPayload & {
  attributes?: {
    [P in keyof ContextPayload]: never;
  } & {
    [key: string]: any;
  };
};

export type LimitedGasScreenEventPayload = GasScreenEventPayload & {
  attributes?: {
    [P in keyof ContextPayload]: never;
  } & {
    [key: string]: any;
  };
};

export const onAdvancedSearchSelected = ({
  trigger,
  actionSubjectId,
  isLoading,
  newTab,
}: AdvancedSearchAnalyticsProps): LimitedGasPayload => {
  return {
    action: 'selected',
    actionSubject: 'advancedSearchLink',
    actionSubjectId,
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'track',
    attributes: {
      trigger,
      isLoading,
      newTab,
    },
  };
};

export interface XButtonAnalyticsProps {
  actionSubjectId: string;
  searchSessionId: string;
}

export const onXButtonClicked = ({
  actionSubjectId,
  searchSessionId,
}: XButtonAnalyticsProps): GasPayloadWithContext => {
  return {
    action: 'dismissed',
    actionSubject: 'stickySearch',
    actionSubjectId,
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'ui',
    attributes: {
      searchSessionId,
    },
  };
};

export const onMultiSiteAdvancedSearchSelected = ({
  trigger,
  actionSubjectId,
  isLoading,
  newTab,
  destinationId,
  index,
}: MultiSiteAdvancedSearchAnalyticsProps): LimitedGasPayload => {
  return {
    action: 'selected',
    actionSubject: 'advancedSearchLink',
    actionSubjectId,
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'ui',
    attributes: {
      trigger,
      isLoading,
      newTab,
      destinationId,
      index,
    },
  };
};

export const onMoreFiltersSelected = ({
  trigger,
  selectedFiltersCount,
}: MoreFiltersAnalyticsProps): LimitedGasPayload => {
  return {
    action: 'selected',
    actionSubject: 'moreFilterLink',
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'ui',
    attributes: {
      trigger,
      selectedFiltersCount,
    },
  };
};

export type ScreenType =
  | 'preQuerySearchResults'
  | 'postQuerySearchResults'
  | 'cachedResults';
export type SectionId = 'page';

export type SectionID =
  | 'confluence-item'
  | 'confluence-space'
  | 'confluence-people'
  | 'jira-issue'
  | 'jira-board-project-filter'
  | string;

export type SearchResultContextPayload = {
  sectionIndex: number;
  sectionId: string;
  containerId: string | null;
  globalIndex: number;
  indexWithinSection: number;
  isCached: boolean;
  metadata?: object;
};

export type OnSearchResultSelectedProps = {
  screen: ScreenType;
  isSticky: boolean;
  isStickyUpdated: boolean;
};

export const onSearchResultSelected = ({
  screen,
  isSticky,
  isStickyUpdated,
}: OnSearchResultSelectedProps): LimitedGasPayload => {
  return {
    action: 'selected',
    actionSubject: 'searchResult',
    actionSubjectId: screen,
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'track',
    attributes: {
      isSticky,
      isStickyUpdated,
    },
  };
};

export type OnSearchResultHighlightedProps = Omit<
  OnSearchResultSelectedProps,
  'isSticky' | 'isStickyUpdated'
>;

export const onSearchResultHighlighted = ({
  screen,
}: OnSearchResultHighlightedProps): LimitedGasPayload => {
  return {
    action: 'highlighted',
    actionSubject: 'searchResult',
    actionSubjectId: screen,
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'ui',
    attributes: {},
  };
};

export enum ResultsShownActionSubjectId {
  PREQUERY = 'preQuerySearchResults',
  POSTQUERY = 'postQuerySearchResults',
  CACHED = 'cachedResults',
}

export interface ResultAnalyticData {
  containerId: string;
  isRecentResult: boolean;
  resultContentId: string;
  resultType: string;
}

export interface ResultSectionAnalyticData {
  results?: ResultAnalyticData[];
  sectionId: SectionID;
}

export interface ResultsShownAnalyticsProps {
  actionSubjectId:
    | 'preQuerySearchResults'
    | 'postQuerySearchResults'
    | 'cachedResults';
  resultCount: number;
  sectionCount: number;
  results: ResultSectionAnalyticData[];
  timeToQueryMs?: number;
  activeProduct: string;
  isMultiProduct: boolean;
  isSticky: boolean;
  isStickyUpdated: boolean;
}

export const onSearchResultsShown = ({
  actionSubjectId,
  resultCount,
  sectionCount,
  results,
  timeToQueryMs,
  activeProduct,
  isMultiProduct,
  isSticky,
  isStickyUpdated,
}: ResultsShownAnalyticsProps): LimitedGasPayload => {
  return {
    action: 'shown',
    actionSubject: 'searchResults',
    actionSubjectId,
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'ui',
    attributes: {
      resultCount,
      sectionCount,
      timeToQueryMs,
      results,
      activeProduct,
      isMultiProduct,
      isSticky,
      isStickyUpdated,
    },
  };
};

export interface TextEnteredAnalyticsProps {
  isSticky: boolean;
}

export const onTextEntered = ({
  isSticky,
}: TextEnteredAnalyticsProps): LimitedGasPayload => {
  return {
    action: 'entered',
    actionSubject: 'text',
    actionSubjectId: 'searchDialogTextField',
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'track',
    attributes: {
      isSticky,
    },
  };
};

export interface OnShowMoreClickedProps {
  total: number;
  currentSize: number;
  pageSize: number;
}

export const onShowMoreClicked = ({
  total,
  currentSize,
  pageSize,
}: OnShowMoreClickedProps): LimitedGasPayload => {
  return {
    action: 'clicked',
    actionSubject: 'button',
    actionSubjectId: 'showMoreButton',
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'ui',
    attributes: {
      total,
      currentSize,
      pageSize,
    },
  };
};

export interface OnShowAllClickedProps {
  total: number;
  currentSize: number;
  pageSize: number;
}

export const onShowAllClicked = ({
  total,
  currentSize,
  pageSize,
}: OnShowMoreClickedProps): LimitedGasPayload => {
  return {
    action: 'clicked',
    actionSubject: 'button',
    actionSubjectId: 'showAllLink',
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'ui',
    attributes: {
      total,
      currentSize,
      pageSize,
    },
  };
};

export interface OnPreQueryScreenViewedProps {
  searchSessionId: string;
  trigger: Trigger;
}

export const onPreQueryScreenViewed = ({
  searchSessionId,
  trigger,
}: OnPreQueryScreenViewedProps): GasScreenEventPayload => {
  return {
    name: 'searchDialogPreQueryScreen',
    eventType: 'screen',
    attributes: {
      searchSessionId,
      trigger,
    },
  };
};

export interface RequestMadeAnalyticsProps {
  actionSubjectId: 'preQuerySearchResults' | 'postQuerySearchResults';
  timings: object;
}

/**
 * We allow onRequestMade to be given an explicit context because it gets fired at the same level as the query contexts so it won't normally pick it up
 */
export const onRequestMade = (
  { actionSubjectId, timings }: RequestMadeAnalyticsProps,
  additionalContext: ContextPayload = {},
  nonPrivacySafeAttributes: NonPrivacySafeContext = { query: '' },
): GasPayloadWithContext => {
  return {
    action: 'completed',
    actionSubject: 'searchRequest',
    actionSubjectId,
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'track',
    attributes: {
      ...timings,
      ...additionalContext,
    },
    nonPrivacySafeAttributes,
  };
};

export const onPreQueryEmpty = (): GasPayloadWithContext => ({
  action: 'shown',
  eventType: 'ui',
  actionSubject: 'searchResults',
  actionSubjectId: 'preQuerySearchResults',
  attributes: {
    resultCount: 0,
    results: [],
  },
  source: BASE_ANALYTICS_SOURCE,
});

export const onDismissed = (): LimitedGasPayload => {
  return {
    action: 'dismissed',
    actionSubject: 'searchDialog',
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'ui',
  };
};

export interface FilterSelectedProps {
  filterId: string;
  filterSource: FilterOptionSource;
  filterType: ProductFilterType;
  isSticky?: boolean;
  isStickyUpdated?: boolean;
}

export const onFilterSelect = ({
  filterId,
  filterType,
  filterSource,
  isSticky,
  isStickyUpdated,
}: FilterSelectedProps): LimitedGasPayload => {
  return {
    action: 'selected',
    actionSubject: 'filter',
    actionSubjectId: 'searchDialogFilter',
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'ui',
    attributes: {
      filterId,
      filterType,
      filterSource,
      isSticky,
      isStickyUpdated,
    },
  };
};

export const onFilterUnselect = ({
  filterId,
  filterType,
  filterSource,
  isSticky,
  isStickyUpdated,
}: FilterSelectedProps): LimitedGasPayload => {
  return {
    action: 'unselected',
    actionSubject: 'filter',
    actionSubjectId: 'searchDialogFilter',
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'ui',
    attributes: {
      filterId,
      filterType,
      filterSource,
      isSticky,
      isStickyUpdated,
    },
  };
};

export const onFiltersCleared = (): LimitedGasPayload => {
  return {
    action: 'clicked',
    actionSubject: 'button',
    actionSubjectId: 'searchDialogClearFilters',
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'ui',
    attributes: {},
  };
};

export const onPaneLoadComplete = (
  product: string,
  duration: number,
): LimitedGasPayload => {
  return {
    action: 'loadComplete',
    actionSubject: 'searchDialogPane',
    actionSubjectId: product,
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'track',
    attributes: {
      durationMs: duration,
    },
  };
};

export const onExperimentLoaded = (
  abTest: ABTest,
  experimentRequestDuration: number,
  isMultiProduct: boolean,
): GasPayloadWithContext => {
  return {
    action: 'loaded',
    actionSubject: 'experiment',
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'operational',
    attributes: {
      abTest,
      experimentRequestDuration,
      isMultiProduct,
    },
  };
};

export const onTabSelected = (
  { tabName }: TabSelectedProps,
  additionalContext: ContextPayload = {},
  nonPrivacySafeAttributes: NonPrivacySafeContext = { query: '' },
): GasPayloadWithContext => {
  // this event is fired at the same level as the anlytics context hence it needs mandatory additional objects.
  return {
    action: 'clicked',
    actionSubject: 'tab',
    actionSubjectId: 'crossProductSearchTab',
    source: BASE_ANALYTICS_SOURCE,
    eventType: 'ui',
    attributes: {
      tabName,
      ...additionalContext,
    },
    nonPrivacySafeAttributes,
  };
};

// Smart User Picker analytics
// These analytics will be used to join with the backend collaboration graph analytics to identify
// if our model is performing well.

export type AtlaskitSelectValue = Option | Array<Option> | null | undefined;

const createEvent = (
  eventType: 'ui' | 'operational',
  action: string,
  actionSubject: string,
  attributes = {},
): LimitedGasPayload => ({
  eventType,
  action,
  actionSubject,
  attributes: {
    ...attributes,
  },
});

const position = (options: OptionData[], value?: Option) => {
  return value
    ? options.findIndex((option) => option.id === value.data.id)
    : -1;
};

const result = (option?: Option) => {
  return option ? optionData2Analytics(option.data) : null;
};

const optionData2Analytics = (option: OptionData) => {
  const { id, type } = option;
  return { id: id ? id : null, type: type || null };
};

const createDefaultPickerAttributes = (
  fieldId: string,
  collabGraphSessionId: string,
) => ({
  context: fieldId,
  isCheckbox: true,
  queryLength: 0,
  collabGraphSessionId,
});

export const selectEvent = (
  value: Option,
  options: OptionData[],
  fieldId: string,
  collabGraphSessionId: string,
) => {
  return createEvent('ui', 'clicked', 'userPicker', {
    ...createDefaultPickerAttributes(fieldId, collabGraphSessionId),
    position: position(options, value),
    result: result(value),
  });
};
