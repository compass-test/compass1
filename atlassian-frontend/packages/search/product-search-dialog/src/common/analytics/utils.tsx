import { createHash } from 'rusha';
import { Trigger } from './events';

export const sha1Hash = (str: string): string =>
  createHash().update(str).digest('hex');

export const getTrigger = (
  e: React.MouseEvent<any, MouseEvent> | MouseEvent,
): Trigger => {
  return e.screenX === 0 ? Trigger.RETURN : Trigger.CLICK;
};

export const isNewTab = (e: React.MouseEvent<any, MouseEvent> | MouseEvent) => {
  return !!(e && (e.metaKey || e.ctrlKey || e.shiftKey));
};

export const getWordCount = (str: string) => {
  return str.length > 0 ? str.split(/\s+/).length : 0;
};
