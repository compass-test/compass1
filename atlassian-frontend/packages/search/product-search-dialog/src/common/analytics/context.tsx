import {
  ContextPayload,
  NonPrivacySafeContext,
  SearchResultContextPayload,
} from '../../common/analytics/events';
import React, { Reducer, useCallback, useReducer } from 'react';
import { getWordCount, sha1Hash } from '../../common/analytics/utils';

interface AdditionalAnalyticsContextState {
  queryVersion: number;
  isLoading: boolean;
}

export type AnalyticsContextAction =
  | { type: 'reset' }
  | {
      type: 'queryVersion';
      value: number;
    }
  | {
      type: 'isLoading';
      value: boolean;
    };

const defaultAnalyticsContextState: AdditionalAnalyticsContextState = {
  queryVersion: 0,
  isLoading: false,
};

export type SupportedContext = SearchResultContextPayload | ContextPayload;

const analyticsContextReducer: Reducer<
  AdditionalAnalyticsContextState,
  AnalyticsContextAction
> = (state, action) => {
  switch (action.type) {
    case 'queryVersion':
      return { ...state, queryVersion: action.value };
    case 'isLoading':
      return { ...state, isLoading: action.value };
    case 'reset':
      return defaultAnalyticsContextState;
    default:
      throw new Error();
  }
};

type AnalyticsHook = (
  query: string,
) => {
  isLoading: boolean;
  setAdditionalAnalyticsContext: React.Dispatch<AnalyticsContextAction>;
  addAnalyticContext: () => ContextPayload;
  nonPrivacySafeContext: () => NonPrivacySafeContext;
  queryVersion: number;
};

export const useAnalyticsContext: AnalyticsHook = (query: string) => {
  const [
    { queryVersion, isLoading },
    setAdditionalAnalyticsContext,
  ] = useReducer(analyticsContextReducer, defaultAnalyticsContextState);
  const addAnalyticContext = useCallback(() => {
    return {
      wordCount: getWordCount(query),
      queryLength: query.length,
      queryHash: sha1Hash(query),
      queryVersion: queryVersion,
    };
  }, [query, queryVersion]);
  const nonPrivacySafeContext = useCallback(() => {
    return { query };
  }, [query]);

  return {
    isLoading,
    setAdditionalAnalyticsContext,
    addAnalyticContext,
    nonPrivacySafeContext,
    queryVersion,
  };
};
