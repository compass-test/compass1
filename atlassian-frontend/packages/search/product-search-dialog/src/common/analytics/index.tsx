import * as ContextImports from './context';
export type SupportedContext = ContextImports.SupportedContext;
export type AnalyticsContextAction = ContextImports.AnalyticsContextAction;
export { SearchDialogAnalyticsContext } from './search-dialog-analytics-context';
export { SearchResultsShownHandler } from './results-shown-handler';
export { withAnalytics } from './with-analytics';
export { useAnalytics } from './use-analytics';
export { DialogDismissedHandler } from './dialog-dismissed-handler';
export { TextEnteredHandler } from './text-entered-handler';
export { useAnalyticsContext } from './context';
export { sha1Hash, getTrigger, isNewTab, getWordCount } from './utils';
export {
  onFilterUnselect,
  onFilterSelect,
  onAdvancedSearchSelected,
  onMultiSiteAdvancedSearchSelected,
  onDismissed,
  onExperimentLoaded,
  onFiltersCleared,
  onMoreFiltersSelected,
  onPaneLoadComplete,
  onPreQueryEmpty,
  onPreQueryScreenViewed,
  onRequestMade,
  onSearchResultHighlighted,
  onSearchResultSelected,
  onSearchResultsShown,
  onShowAllClicked,
  onShowMoreClicked,
  onTabSelected,
  onTextEntered,
  AdvancedSearchLinkSubjectId,
  selectEvent,
} from './events';

export type {
  ScreenType,
  NonPrivacySafeContext,
  OnPreQueryScreenViewedProps,
  OnSearchResultHighlightedProps,
  OnSearchResultSelectedProps,
  OnShowAllClickedProps,
  OnShowMoreClickedProps,
  SearchResultContextPayload,
  SectionID,
  ResultAnalyticData,
  ResultSectionAnalyticData,
} from './events';
