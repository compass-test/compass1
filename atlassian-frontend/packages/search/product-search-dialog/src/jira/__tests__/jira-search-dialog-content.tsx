import React from 'react';

import { shallow } from 'enzyme';

import SearchDialogContent from '../jira-search-dialog-content';
import { AnalyticsContextAction } from '../../common/analytics';

jest.mock('@atlaskit/width-detector', () => ({
  WidthObserver: () => null,
}));

describe('<SearchDialogContent />', () => {
  const props = {
    isAnyResultsLoading: false,
    isExpanded: true,
    jiraFeatures: { hasSoftwareAccess: false },
    setNoResultsForPreQuery: (isLoading: boolean) => isLoading,
    setAdditionalAnalyticsContext: (action: AnalyticsContextAction) => action,
    query: 'query',
    onRetry: () => undefined,
    onRequestComplete: (timings: number) => timings,
    onNavigate: () => {},
    queryVersion: 0,
    debounceTime: 0,
  };

  const getWrapper = (passedProps = props) =>
    shallow(<SearchDialogContent {...passedProps} />);

  it('should match the snapshot', () => {
    expect(getWrapper()).toMatchSnapshot();
  });
});
