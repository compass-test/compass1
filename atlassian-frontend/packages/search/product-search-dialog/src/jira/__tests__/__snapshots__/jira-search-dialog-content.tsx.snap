// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchDialogContent /> should match the snapshot 1`] = `
<Fragment>
  <SearchDialogWithContext
    debounceTime={0}
    isAnyResultsLoading={false}
    isExpanded={true}
    jiraFeatures={
      Object {
        "hasSoftwareAccess": false,
      }
    }
    onNavigate={[Function]}
    onRequestComplete={[Function]}
    onRetry={[Function]}
    query="query"
    queryVersion={0}
    setAdditionalAnalyticsContext={[Function]}
    setNoResultsForPreQuery={[Function]}
  />
  <WarmUpFiltersCache />
</Fragment>
`;
